cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -584,18 +590,22 @@ struct get_page_span {
     auto const num_pages         = column_page_end - column_page_start;
     bool const is_list           = chunks[column_index].max_level[level_type::REPETITION] > 0;
 
-    auto start_page =
-      (thrust::lower_bound(thrust::seq, column_page_start, column_page_end, start_row) -
-       column_page_start) +
-      first_page_index;
-    // list rows can span page boundaries, so it is not always safe to assume that the row
-    // represented by end_row_index starts on the subsequent page. It is possible that
-    // the values for row end_row_index start within the page itself. so we must
-    // include the page in that case.
-    if (page_row_index[start_page] == start_row && !is_list) { start_page++; }
-
-    auto end_page = (thrust::lower_bound(thrust::seq, column_page_start, column_page_end, end_row) -
-                     column_page_start) +
+    // For list columns, the row counts are estimates so we need all prefix pages to correctly
+    // compute page bounds. For non-list columns, we can get an exact span of pages.
+    auto start_page              = first_page_index;
+    auto const update_start_page = has_page_index or (not is_list) or (not is_first_subpass);
```

Comment:
With this change, we may need to select at most `K-1` extra pages where `K` is the number of data pages in the first row group of the first pass. 

Note: Although `is_first_subpass` will be true for the first subpass of subsequent passes but that won't add any pages that we don't need since `skip_rows/num_rows` trim the pages from either termini (start or end) and not in between.

Reply:
Also, if page_index is present, the `page.num_rows` is accurate (even for lists as list rows can't span pages if page index is present) so we can just use this fact to trim pages confidently.